### ISSUE TITLE
[3.0.4] DNSServiceDiscovery 出现 metaService NPE

### ISSUE THREAD

**Issue by Sivan757** [apache/dubbo#9493]
- [x] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 3.0.4、3.0.5-SNAPSHOT
* Operating System version: macos 12.1
* Java version: 11

### Steps to reproduce this issue

1. 运行 dubbo-samples 中的 [dubbo-samples-kubernetes-dns](https://github.com/apache/dubbo-samples/tree/master/dubbo-samples-kubernetes/dubbo-samples-kubernetes-dns) provider启动类

Pls. provide [GitHub address] to reproduce this issue.

### Expected Behavior

dns 进行服务发现

### Actual Behavior

metaservice为空 抛出空指针异常后，正常启动，但导致消费者无法通过 MetadataService 获取服务信息，导致消费者调用失败

查看源码发现 metadataService 通过触发 setApplicationModel() 后加载（在 3.0.2.1 中为直接获取默认拓展）。

尝试`dubbo.application.metadata-type=local` 后，`SelfHostMetaServiceDiscovery#setApplicationModel()`依然没有被触发。

```java
java.lang.NullPointerException
	at org.apache.dubbo.registry.client.SelfHostMetaServiceDiscovery.updateMetadata(SelfHostMetaServiceDiscovery.java:164)
	at org.apache.dubbo.registry.client.SelfHostMetaServiceDiscovery.register(SelfHostMetaServiceDiscovery.java:187)
	at org.apache.dubbo.registry.client.metadata.ServiceInstanceMetadataUtils.lambda$registerMetadataAndInstance$1(ServiceInstanceMetadataUtils.java:264)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.apache.dubbo.registry.client.metadata.ServiceInstanceMetadataUtils.registerMetadataAndInstance(ServiceInstanceMetadataUtils.java:255)
	at org.apache.dubbo.config.deploy.DefaultApplicationDeployer.registerServiceInstance(DefaultApplicationDeployer.java:744)
	at org.apache.dubbo.config.deploy.DefaultApplicationDeployer.prepareApplicationInstance(DefaultApplicationDeployer.java:624)
	at org.apache.dubbo.config.deploy.DefaultModuleDeployer.start(DefaultModuleDeployer.java:142)
	at org.apache.dubbo.config.spring.context.DubboDeployApplicationListener.onContextRefreshedEvent(DubboDeployApplicationListener.java:108)
	at org.apache.dubbo.config.spring.context.DubboDeployApplicationListener.onApplicationEvent(DubboDeployApplicationListener.java:98)
	at org.apache.dubbo.config.spring.context.DubboDeployApplicationListener.onApplicationEvent(DubboDeployApplicationListener.java:44)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:393)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:347)
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:883)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:546)
	at org.springframework.context.annotation.AnnotationConfigApplicationContext.<init>(AnnotationConfigApplicationContext.java:84)
	at org.apache.dubbo.samples.ProviderBootstrap.main(ProviderBootstrap.java:32)
```

**Comment by chickenlj** [comment]
We will try to fix it in the next 3.0.6 version.

**Comment by Sivan757** [comment]
> We will try to fix it in the next 3.0.6 version.

Thanks, but I don’t know, maybe I can modify something to make it work on 3.0.4?

**Comment by CrazyHZM** [comment]
transfer to https://github.com/apache/dubbo-spi-extensions
